### PR TITLE
libopaevfio: add interrupt support

### PR DESCRIPTION
**Pull request by tswhison**
Signed-off-by: Tim Whisonant <tim.whisonant@intel.com>

### CODE REVIEW

File: libopaevfio/opaevfio.c [hunk 1]
```diff
@@ -756,6 +807,133 @@ int opae_vfio_buffer_free(struct opae_vfio *v,
 	return res;
 }
 
+int opae_vfio_irq_enable(struct opae_vfio *v,
+                         uint32_t index,
+                         uint32_t subindex,
+                         int event_fd)
+{
```
Comment (anandaravuri) [inline libopaevfio/opaevfio.c:814]: IRQ trigger code need to added in enable / disable

https://inbox.dpdk.org/dev/1796500.5oFe8j95cd@xps/T/

https://gitea.ananwork.com/mirrors/dpdk/commit/5c782b3928b8

File: libopaevfio/opaevfio.c [hunk 2]
```diff
@@ -756,6 +807,133 @@ int opae_vfio_buffer_free(struct opae_vfio *v,
 	return res;
 }
 
+int opae_vfio_irq_enable(struct opae_vfio *v,
+                         uint32_t index,
```
Comment (anandaravuri) [inline libopaevfio/opaevfio.c:811]: event_fd should be array of fd ?

File: libopaevfio/opaevfio.c (hunk 1 above)
Comment (anandaravuri) [reply libopaevfio/opaevfio.c:814]: vfio_enable_msix() vfio_dsiable_msix() example

Comment (tswhison) [reply libopaevfio/opaevfio.c:814]: > IRQ trigger code need to added in enable / disable
> 
> https://inbox.dpdk.org/dev/1796500.5oFe8j95cd@xps/T/
> 
> https://gitea.ananwork.com/mirrors/dpdk/commit/5c782b3928b8

Are you referring to VFIO_IRQ_SET_ACTION_TRIGGER?

Comment (anandaravuri) [reply libopaevfio/opaevfio.c:814]: yes, action trigger

**Review by anandaravuri — APPROVED** [review]

**Pull request merged**